### ISSUE TITLE
useHead with reactive script src and key leads to script-tag vanishing instead of being updated

### ISSUE THREAD

**Issue by codeflorist** [nuxt/nuxt#20129]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.3.3`
- Nitro Version:    `2.3.2`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-7fwlmq?file=app.vue

### Describe the bug

consider this setup to make a script-tag reactive per page:

`app.vue`:
```js
useHead({
  script: [
    {
      key: 'TestScript',
      src: () => route.meta.test,
    },
  ],
});
```

`index.vue`, `page.vue`, etc.:
```js
definePageMeta({
  test: 'page-test',
});
```

this leads to the script tag vanishing instead of getting updated on navigating to a new page.

removing the `key` solves the problem, but it is needed in this usecase to avoid duplication.

furthermore, this does not happen with `meta` tags, but only with `script` tags.

**Comment by danielroe** [comment]
cc: @harlan-zw

**Comment by harlan-zw** [comment]
Hey @codeflorist, thanks for the issue.

Can confirm this is a bug. Will try and fix it in the next couple of days.

**Comment by harlan-zw** [comment]
This should be fixed in Unhead [v1.1.26](https://github.com/unjs/unhead/releases/tag/v1.1.26), you can delete lock file + re-install to upgrade.

**Comment by codeflorist** [comment]
impeccable. thanks a lot! 😊👍
